src/dev/src/app/tree-view/eager-recursive-tree/eager-recursive-tree.ts

```diff
@@ -41,12 +44,25 @@ export class EagerRecursiveTreeDemo {
     },
   ];
 
-  singleRootSelected: { [key: string]: ClrSelectedState } = {};
-  multiRootSelected: { [key: string]: ClrSelectedState } = {};
+  private buildDefaultSelected(rootMap, selectedMap: SelectedMap = {}) {
+    if (!Array.isArray(rootMap)) {
+      rootMap = [rootMap];
```

Comment:
In this case of `!Array.isArray(rootMap)`, `rootMap` is `nodelist`, no? Shouldn't we do `Array.from(rootMap)` instead of `[rootMap]`?

Reply:
`rootMap` is either an array or an object, see the `singleRoot` and `multiRoot` properties on the class.

Reply:
But on line 51, we have `this.buildDefaultSelected(node.children, selectedMap);` 
So we are basically putting `node.children`, which is `NodeList` in the array.